File: LLM-Translations/GPT-4/codenet/Java/Go/s443066739.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())

	pre := make([]int64, n+1)

	scanner.Scan()
	line := scanner.Text()
	nums := strings.Fields(line)

	for i := 1; i <= n; i++ {
		num, _ := strconv.ParseInt(nums[i-1], 10, 64)
		pre[i] = pre[i-1] + num
	}

	min := int64(math.MaxInt64)

	for i := 1; i < n; i++ {
		val1 := pre[i]
		val2 := pre[n] - pre[i]

		min = int64(math.Min(float64(min), math.Abs(float64(val1-val2))))
	}

	fmt.Println(min)
}
// 

File: LLM-Translations/GPT-4/codenet/Java/C/s443066739.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <limits.h>

int main()
{
    int n;
    scanf("%d", &n);
    long long pre[n+1];
    pre[0] = 0;

    for(int i=1; i<=n; i++)
    {
        long long val;
        scanf("%lld", &val);
        pre[i] = pre[i - 1] + val;
    }

    long long min = LLONG_MAX;

    for(int i=1; i<n; i++)
    {
        long long val1 = pre[i], val2 = pre[n] - pre[i];
        long long diff = llabs(val1 - val2);
        if(diff < min)
        {
            min = diff;
        }
    }

    printf("%lld\n", min);
    
    return 0;
}
// 
